pallet/account-migration/src/lib.rs

```diff
@@ -123,10 +127,10 @@ pub mod pallet {
 	/// [`pallet_asset::AssetAccount`] data.
 	///
 	/// https://github.dev/paritytech/substrate/blob/polkadot-v0.9.30/frame/assets/src/types.rs#L115
+	// The size of `pallet_asset::AssetAccount` is 64 bytes.
 	#[pallet::storage]
 	#[pallet::getter(fn kton_account_of)]
-	pub type KtonAccounts<T: Config> =
-		StorageMap<_, Blake2_128Concat, AccountId32, AssetAccount<u128, u128, ()>>;
+	pub type KtonAccounts<T: Config> = StorageMap<_, Blake2_128Concat, AccountId32, [u8; 64]>;
```

Comment:
Why did the `Extra` field change to `[u8; 64]`?

```rs
pub struct AssetAccount<Balance, DepositBalance, Extra> {
	/// The balance.
	pub(super) balance: Balance,
	/// Whether the account is frozen.
	pub(super) is_frozen: bool,
	/// The reason for the existence of the account.
	pub(super) reason: ExistenceReason<DepositBalance>,
	/// Additional "sidecar" data, in case some other pallet wants to use this storage item.
	pub(super) extra: Extra,
}
```

The above `AssetAccount` is used in storage like this. This third parameter comes from runtime `<T as Config<I>>::Extra`.

```rs
pub(super) type AssetAccountOf<T, I> =
	AssetAccount<<T as Config<I>>::Balance, DepositBalanceOf<T, I>, <T as Config<I>>::Extra>;
```

However in the darwinia's runtime:

```rs
impl pallet_assets::Config for Runtime {
	...
	type Extra = ();
	...
}

```
I think it should be `()` here.

Reply:
O, you replace the `AssetAccount` type with a bytes array.